### PR TITLE
YUNIKORN-1390 - MPIOperator Example to Yunikorn Site

### PR DESCRIPTION
**Pull request by kannon92**
This is a WIP due to https://github.com/apache/yunikorn-k8shim/pull/478.

### CODE REVIEW

**Comment by wilfred-s** [comment]
Is this still WIP or can we go ahead and get both the k8shim and this one committed indepently. I doubt there will be much delay between committing the example change and this website update

**Comment by kannon92** [comment]
> Is this still WIP or can we go ahead and get both the k8shim and this one committed indepently. I doubt there will be much delay between committing the example change and this website update

It’s almost done. I think I have some link issues that I have to fix. I put it in WIP mostly because I wasn’t sure how long the other PR would take.

**Comment by kannon92** [comment]
PR is ready for review.

File: docs/user_guide/workloads/run_mpi.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+
+# MPI Operator
+
+For more details, please read https://github.com/kubeflow/mpi-operator#installation for deploying this operator.
```
Comment (wilfred-s) [inline docs/user_guide/workloads/run_mpi.md:31]: Can we get this more inline with the [tensorflow](https://yunikorn.apache.org/docs/user_guide/workloads/run_tf) doc we have:
* Run MPI Jobs
* Install mpi-operator
Followed by the rest of the page. That would give a proper attribution of the mpi operator to kubeflow etc

**Review by wilfred-s — CHANGES_REQUESTED** [review]

File: docs/user_guide/workloads/run_mpi.md (hunk 1 above)
Comment (0yukali0) [inline docs/user_guide/workloads/run_mpi.md:31]: Change https://github.com/kubeflow/mpi-operator#installation to
[Installation steps of mpi-operator](https://github.com/kubeflow/mpi-operator#installation)

File: docs/user_guide/workloads/run_mpi.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            args:
+            - -De
+            - -f
+            - /home/mpiuser/.sshd_config
+            resources:
+              limits:
+                cpu: 1
+                memory: 1Gi
+```
+```
```
Comment (0yukali0) [inline docs/user_guide/workloads/run_mpi.md:100]: Add '## Run MPI Jobs' to separate the pi.yaml and the kubectl command.

**Review by 0yukali0 — COMMENTED** [review]
Hi @kannon92, this doc will not be found when users visit the website.
Adding 'user_guide/workloads/run_mpi' to the yunikorn_site/sidebar.js file is necessary.
The paragraph of sidebar.js would look like.
```Part of sidebar.js
 items: [
                    'user_guide/workloads/workload_overview',
                    'user_guide/workloads/run_spark',
                    'user_guide/workloads/run_flink',
                    'user_guide/workloads/run_tf',
                    'user_guide/workloads/run_mpi'
                ],
```

**Comment by kannon92** [comment]
Thank you for your feedback! I’m at a conference this week but I will address these comments soon.

File: docs/user_guide/workloads/run_mpi.md (hunk 1 above)
Comment (kannon92) [reply docs/user_guide/workloads/run_mpi.md]: I think I did what you asked.  Please look it over when you get a chance.

Comment (kannon92) [reply docs/user_guide/workloads/run_mpi.md]: Applied both suggestions!

File: docs/user_guide/workloads/run_mpi.md (hunk 2 above)
Comment (wilfred-s) [reply docs/user_guide/workloads/run_mpi.md:105]: As @0yukali0 mentioned: yaml and kubectl separation is needed to clarify what is documented. You can use '## Example MPI job yaml` and '## Run MPI Jobs'

Comment (kannon92) [reply docs/user_guide/workloads/run_mpi.md:106]: Following the tensor flow docs, I notice you just added a simple text to separate the yaml from the kubectl code.  I figured that would be fine here.  

If you want a section, I can add that also.

File: sidebars.js [hunk 3]
```diff
@@ -37,6 +37,7 @@ module.exports = {
                     'user_guide/workloads/workload_overview',
                     'user_guide/workloads/run_spark',
                     'user_guide/workloads/run_flink',
+                    'user_guide/workflows/run_mpi',
```
[the remark this answers is not in the document]
Comment (kannon92) [reply sidebars.js]: Thank you for telling me that!

[the remark this answers is not in the document]
Comment (kannon92) [reply sidebars.js]: I updated this change.

**Review by wilfred-s — APPROVED** [review]
LGTM
one nit: keep the order in the `sidebars.js` and `workload_overview.md` the same for the loads. That can be fixed as part of the commit.

**Review by 0yukali0 — COMMENTED** [review]
As @wilfred-s  said, keeping the order among workload and sidebars consistent would be better.

**Review by 0yukali0 — APPROVED** [review]
LGTM

**Pull request merged**